Sources/Commands/SwiftTool.swift

```diff
@@ -571,6 +586,7 @@ public class SwiftTool {
                 sharedConfigurationDirectory: sharedConfigurationDirectory
             ),
             mirrors: self.getMirrorsConfig(sharedConfigurationDirectory: sharedConfigurationDirectory).mirrors,
+            registries: try? self.getRegistriesConfig(sharedConfigurationDirectory: sharedConfigurationDirectory).configuration,
```

Comment:
`try?` intentional? should it throw instead?

Reply:
If we throw here, package resolution would fail for anyone who doesn't have a `.swiftpm/configuration/registries.json` file, even if they only use URL-based dependencies. The check for valid registry configuration is pushed back to be done only when necessary.